tests/rule_based_profiler/test_rule_based_profiler.py

```diff
@@ -0,0 +1,399 @@
+                },
+            ],
+            "expectation_configuration_builders": [
+                {
+                    "expectation_type": "expect_column_pair_values_A_to_be_greater_than_B",
+                    "meta": {},
+                },
+                {
+                    "expectation_type": "expect_column_min_to_be_between",
+                    "meta": {},
+                },
+            ],
+        },
+        {
+            "name": "rule_1",
+            "domain_builder": {},
+            "parameter_builders": [
+                {
+                    "name": "my_parameter",
+                    "metric_name": "my_metric",
+                    "enforce_numeric_metric": False,
+                    "replace_nan_with_zero": False,
+                },
+            ],
+            "expectation_configuration_builders": [
+                {
+                    "expectation_type": "expect_column_pair_values_A_to_be_greater_than_B",
+                    "meta": {},
+                },
+            ],
+        },
+    ]
+    effective_rules: List[
+        Rule
+    ] = profiler_with_placeholder_args.reconcile_profiler_rules(rules=rules)
+    rule_name: str
+    rule: Rule
+    rule_config: dict
+    assert {rule.name: rule.to_json_dict() for rule in effective_rules} == {
+        rule_config["name"]: rule_config for rule_config in expected_rules
+    }
+
+
+def test_reconcile_profiler_rules_existing_rule_domain_builder_override(
+    profiler_with_placeholder_args,
+):
+    rules: Dict[str, Dict[str, Any]] = {
+        "rule_1": {
+            "domain_builder": {
+                "class_name": "SimpleColumnSuffixDomainBuilder",
+                "module_name": "great_expectations.rule_based_profiler.domain_builder",
+                "column_name_suffixes": [
+                    "_ts",
+                ],
+            },
+        },
+    }
+
+    expected_rules: List[dict] = [
+        {
+            "name": "rule_1",
+            "domain_builder": {
+                "column_name_suffixes": [
+                    "_ts",
+                ],
+            },
+            "parameter_builders": [
+                {
+                    "name": "my_parameter",
+                    "metric_name": "my_metric",
+                    "enforce_numeric_metric": False,
+                    "replace_nan_with_zero": False,
+                },
+            ],
+            "expectation_configuration_builders": [
+                {
+                    "expectation_type": "expect_column_pair_values_A_to_be_greater_than_B",
+                    "meta": {},
+                },
+            ],
+        },
+    ]
+    effective_rules: List[
+        Rule
+    ] = profiler_with_placeholder_args.reconcile_profiler_rules(rules=rules)
+    rule_name: str
+    rule: Rule
+    rule_config: dict
+    assert {rule.name: rule.to_json_dict() for rule in effective_rules} == {
+        rule_config["name"]: rule_config for rule_config in expected_rules
+    }
```

Comment:
Same note here.

Reply:
@cdkini I will fix this once we get past the `Python-3.6 related failures` and before merging.  Thanks!